agent/agent.go

```diff
@@ -191,16 +191,26 @@ func _main() int {
 		taskEngine = engine.NewTaskEngine(cfg, dockerClient, credentialsManager, containerChangeEventStream, imageManager, state)
 	}
 
-	// Load Pause Container Image
-	_, err = pause.LoadImage(cfg, dockerClient)
-	if err != nil {
-		if !pause.UnsupportedPlatform(err) {
-			log.Criticalf("Error loading pause container image: %v", err)
+	if cfg.TaskENIEnabled {
+		// Load Pause Container Image
+		_, err = pause.LoadImage(cfg, dockerClient)
+		if err != nil {
+			if !pause.UnsupportedPlatform(err) {
+				log.Criticalf("Error loading pause container image: %v", err)
+				return exitcodes.ExitError
```

Comment:
I think we should be a bit more nuanced here. We should let LoadImage return `NoSuchFileError` (`"no such file or directory"`) [here](https://github.com/aws/amazon-ecs-agent/blob/enis/agent/eni/pause/pause_linux.go#L44)and `DockerError` and handle these differently. If it's `NoSuchFileError` , it should be a terminal error, otherwise it's not.